clippy_lints/src/cargo/mod.rs

```diff
@@ -155,12 +155,25 @@ declare_clippy_lint! {
     /// [As the edition guide says](https://rust-lang-nursery.github.io/edition-guide/rust-2018/cargo-and-crates-io/crates-io-disallows-wildcard-dependencies.html),
     /// it is highly unlikely that you work with any possible version of your dependency,
     /// and wildcard dependencies would cause unnecessary breakage in the ecosystem.
+    /// Instead, you must actually specify a specific version or range of versions, using one of the `semver` crate’s various options: `^`, `~`, or `=`.
     ///
     /// ### Example
     /// ```toml
     /// [dependencies]
     /// regex = "*"
     /// ```
+    /// Use instead:
+    /// ```toml
+    /// [dependencies]
+    /// # compatible with the specified version and any version that doesn't introduce breaking changes
+    /// some_crate_1 = "^1.2.3"
```

Comment:
The `^` part isn't necessary. It's what cargo does with a bare version number.

Reply:
Can you add this part of the example back in, just without the `^` as part of the version number. I would word it as any semver compatible version rather than any version without breaking changes.